### PR TITLE
Add postal code validation to concern

### PR DESCRIPTION
**Pull request by hez**
- renamed to postal_codeable as it does more now
- validates formatting with something sane-ish
- upper cases before saving

### CODE REVIEW

**Comment by hez** [comment]
I ended up jamming this in with the geolocation stuff, if we don't like that it could be separated.

File: app/models/concerns/postal_codeable.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    # Stolen from https://github.com/globaldev/going_postal/blob/master/lib/going_postal.rb
+    unless defined?(POSTAL_FIRST_PART)
+      POSTAL_FIRST_PART = /[A-CEGHJK-NPR-TVXY][0-9][A-CEGHJK-NPR-TV-Z]/.freeze
+    end
+    unless defined?(POSTAL_SECOND_PART)
+      POSTAL_SECOND_PART = /[0-9][A-CEGHJK-NPR-TV-Z][0-9]/.freeze
+    end
+    unless defined?(CANADIAN_POSTAL_REGEX)
+      CANADIAN_POSTAL_REGEX =
+        /\A#{POSTAL_FIRST_PART}.?#{POSTAL_SECOND_PART}\z/.freeze
```
Comment (Baradoy) [inline app/models/concerns/postal_codeable.rb]: I am curious about the `.?`, shouldn't that be a `\s?`?

Comment (hez) [reply app/models/concerns/postal_codeable.rb]: I think Postal Codes come in "XXX XXX" "XXX-XXX" and "XXXXXX" ... I just figured that was the easiest... I'm going to write a few specs for the Postal Codeable concern though 👍

**Comment by hez** [comment]
I don't know if the forms do yet, this adds them to the regular AR model errors. That will be the next step.

**Comment by rfb** [comment]
looks great :+1:

**Pull request merged**